src/system.jl

```diff
@@ -0,0 +1,329 @@
+    zones::Dictionary{Int, Zone}
+    "Buses in the `System` indexed by bus name"
+    buses::Dictionary{InlineString15, Bus}
+    "Generators in the `System` indexed by unit code"
+    generators::Dictionary{Int, Generator}
+    "Branches in the `System` indexed by branch name"
+    branches::Dictionary{InlineString31, Branch}
+    """
+    The line outage distribution factor matrix of the system for a set of contingencies given
+    by the keys of the `Dict`. Each entry is a `KeyedArray` with axis keys
+    `branch names x branch on outage`
+    """
+    LODF::Dict{String, KeyedArray{Float64, 2}}
+    """
+    Power transfer distribution factor of the system.  `KeyedArray` where the axis keys are
+    `branch names x bus names`
+    """
+    PTDF::KeyedArray{Float64, 2}
+
+    # Generator related time series
+    "Generation of the generators at the start of the day (MWs)"
+    initial_generation::KeyedArray{Float64, 1}
+    "Hours each generator has been at its current status at the start of the day"
+    hours_at_status::KeyedArray{Float64, 1}
+    "Generator offer curves. `KeyedArray` where the axis keys are `generator names x datetimes`"
+    offer_curve::KeyedArray{Vector{Tuple{Float64, Float64}}, 2}
+    "Generator availability"
+    availability::KeyedArray{Bool, 2}
+    "Generator must run flag indicating that the generator has to be committed at that hour"
+    must_run::KeyedArray{Bool, 2}
+    "Generator minimum output in the ancillary services market (MWs)"
+    regulation_min::KeyedArray{Float64, 2}
+    "Generator maximum output in the ancillary services market (MWs)"
+    regulation_max::KeyedArray{Float64, 2}
+    "Generator minimum output (MWs)"
+    pmin::KeyedArray{Float64, 2}
+    "Generator maximum output (MWs)"
+    pmax::KeyedArray{Float64, 2}
+    "Ancillary services regulation offer prices (\$ /MW)"
+    asm_regulation::KeyedArray{Float64, 2}
+    "Ancillary services spinning offer prices (\$ /MW)"
+    asm_spin::KeyedArray{Float64, 2}
+    "Ancillary services supplemental on offer prices (\$ /MW)"
+    asm_sup_on::KeyedArray{Float64, 2}
+    "Ancillary services supplemental off offer prices (\$ /MW)"
+    asm_sup_off::KeyedArray{Float64, 2}
+
+    # Load time series
+    "Load time series data. `KeyedArray` where the axis keys are `load ids x datetimes`"
+    loads::KeyedArray{Float64, 2}
+
+    # Virtuals/PSD time series
+    "Increment bids time series data. `KeyedArray` where the axis keys are `bid ids x datetimes`"
+    increment_bids::KeyedArray{Vector{Tuple{Float64, Float64}}, 2}
+    "Decrement bids time series data. `KeyedArray` where the axis keys are `bid ids x datetimes`"
+    decrement_bids::KeyedArray{Vector{Tuple{Float64, Float64}}, 2}
+    "Price sensitive demand time series data. `KeyedArray` where the axis keys are `bid ids x datetimes`"
+    price_sensitive_demand::KeyedArray{Vector{Tuple{Float64, Float64}}, 2}
```

Comment:
Being a bit pedantic but PSDs are also bids, so it sounds weird to have `increment_bids`, `decrement_bids`, and `price_sensitive_demand`. Perhaps just remove the `_bids` suffix altogether to avoid long names?